specification/sbi_atee.adoc

```diff
@@ -938,6 +938,202 @@ The possible error codes returned in `sbiret.error` are shown below.
 | SBI_ERR_FAILED          | The operation failed for unknown reasons.
 |===
 
+[#sbi_tee_host_tvm_block_pages]
+=== Function: TEE Host TVM Block Pages (FID #16)
+[source, C]
+-----
+struct sbiret sbi_tee_host_tvm_block_pages(unsigned long tvm_guest_id,
+                                           unsigned long guest_page_address,
+                                           unsigned long length);
+-----
+
+Invalidates the pages in the specified range of guest physical address space and
+marks the page as either `blocked` or `BlockedShared`.
+
+For each page in the range, the TSM must verify that:
+
+* The page is currently marked present in the TVM’s page table.
+* The page is in either the `Mapped` state and uniquely owned by the TVM, or in the
+`Shared` state and owned by the host.
+
+After verifying these pre-conditions are met, the TSM:
+
+* Invalidates the page.
+* Places the page in the `Blocked` state (if `Mapped`) or `BlockedShared` state (if
+`Shared`) and records the TVM TLB version at which the page was invalidated. If the
+page was `Shared`, the reference count of the page is recorded as well.
+
+Guest page faults taken by the TVM on blocked pages continue to be reported to the host.
+The page remains invalid until the mapping is unblocked (marked present), removed, or
+becomes part of a huge page by promotion/demotion operation.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_host_tvm_block_pages_errors]
+.TEE Host TVM Block Pages
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_PARAM   | `tvm_guest_id` or `guest_page_address` or `length` were invalid.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+[#sbi_tee_host_tvm_unblock_pages]
+=== Function: TEE Host TVM Unblock Pages (FID #17)
+[source, C]
+-----
+struct sbiret sbi_tee_host_tvm_unblock_pages(unsigned long tvm_guest_id,
+                                           unsigned long base_page_address,
```

Comment:
instead of base_page_address, can we call it gpa which is concise and clearly indicate what it is

Reply:
Commits removed. I will be opening a new PR for that after reworking the share/unshare parts.